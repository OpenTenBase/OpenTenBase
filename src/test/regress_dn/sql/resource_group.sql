select oid, * from pg_resgroup;
select * from pg_resgroupcapability;
create resource group rg1 with (concurrency=3, cpu_rate_limit=20, memory_limit=30);
create resource group rg1 with concurrency 3 cpu_rate_limit 20 memory_limit 30;
select count(*) from pg_resgroup;
select count(*) from pg_resgroupcapability;
select count(*) from pg_authid;
create role user1 with resource group rg1 password 'pass1';
select count(*) from pg_authid;
select rolname, rolresgroup, rolpriority from pg_authid where rolname = 'user1';
alter role user1 with resource group none priority 'high';
select rolname, rolresgroup, rolpriority from pg_authid where rolname = 'user1';
drop resource group rg1;
select count(*) from pg_resgroup;
select count(*) from pg_resgroupcapability;
drop role user1;
select count(*) from pg_authid;

-- Test: cpuset cannot be specified when group is disabled.
-- CREATE RESOURCE GROUP resource_group1 WITH (memory_limit=5, cpuset='0');
-- CREATE RESOURCE GROUP resource_group1 WITH (memory_limit=5, cpu_rate_limit=5);
-- ALTER RESOURCE GROUP resource_group1 SET cpuset '0';

-- DROP RESOURCE GROUP resource_group1;

-- Test: check gucs
show enable_resource_group;
show resource_group_cpu_priority;
show resource_group_cpu_limit;
show resource_group_memory_limit;
show resource_group_cpu_ceiling_enforcement;
show resource_group_bypass;
show resource_group_bypass;
