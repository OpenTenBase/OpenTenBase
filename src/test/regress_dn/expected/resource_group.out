select oid, * from pg_resgroup;
 oid  |    rsgname    | parent 
------+---------------+--------
 9019 | default_group |      0
 9020 | admin_group   |      0
(2 rows)

select * from pg_resgroupcapability;
 resgroupid | reslimittype | value 
------------+--------------+-------
       9019 |            1 | 20
       9019 |            2 | 30
       9019 |            3 | -1
       9019 |            4 | -1
       9020 |            1 | 10
       9020 |            2 | 10
       9020 |            3 | -1
       9020 |            4 | -1
(8 rows)

create resource group rg1 with (concurrency=3, cpu_rate_limit=20, memory_limit=30);
ERROR:  syntax error at or near "("
LINE 1: create resource group rg1 with (concurrency=3, cpu_rate_limi...
                                       ^
create resource group rg1 with concurrency 3 cpu_rate_limit 20 memory_limit 30;
WARNING:  resource group is disabled
HINT:  To enable set enable_resource_group=true
select count(*) from pg_resgroup;
 count 
-------
     3
(1 row)

select count(*) from pg_resgroupcapability;
 count 
-------
    12
(1 row)

select count(*) from pg_authid;
 count 
-------
     9
(1 row)

create role user1 with resource group rg1 password 'pass1';
WARNING:  resource group is disabled
HINT:  To enable set enable_resource_group=true
select count(*) from pg_authid;
 count 
-------
    10
(1 row)

select rolname, rolresgroup, rolpriority from pg_authid where rolname = 'user1';
 rolname | rolresgroup | rolpriority 
---------+-------------+-------------
 user1   | rg1         | low
(1 row)

alter role user1 with resource group none priority 'high';
WARNING:  resource group is disabled
HINT:  To enable set enable_resource_group=true
select rolname, rolresgroup, rolpriority from pg_authid where rolname = 'user1';
 rolname |  rolresgroup  | rolpriority 
---------+---------------+-------------
 user1   | default_group | high
(1 row)

drop resource group rg1;
select count(*) from pg_resgroup;
 count 
-------
     2
(1 row)

select count(*) from pg_resgroupcapability;
 count 
-------
     8
(1 row)

drop role user1;
select count(*) from pg_authid;
 count 
-------
     9
(1 row)

-- Test: cpuset cannot be specified when group is disabled.
-- CREATE RESOURCE GROUP resource_group1 WITH (memory_limit=5, cpuset='0');
-- CREATE RESOURCE GROUP resource_group1 WITH (memory_limit=5, cpu_rate_limit=5);
-- ALTER RESOURCE GROUP resource_group1 SET cpuset '0';
-- DROP RESOURCE GROUP resource_group1;
-- Test: check gucs
show enable_resource_group;
 enable_resource_group 
-----------------------
 off
(1 row)

show resource_group_cpu_priority;
 resource_group_cpu_priority 
-----------------------------
 10
(1 row)

show resource_group_cpu_limit;
 resource_group_cpu_limit 
--------------------------
 0.9
(1 row)

show resource_group_memory_limit;
ERROR:  unrecognized configuration parameter "resource_group_memory_limit"
show resource_group_cpu_ceiling_enforcement;
 resource_group_cpu_ceiling_enforcement 
----------------------------------------
 off
(1 row)

show resource_group_bypass;
 resource_group_bypass 
-----------------------
 off
(1 row)

show resource_group_bypass;
 resource_group_bypass 
-----------------------
 off
(1 row)

